cmd/fpga_crihook/main.go

```diff
@@ -124,8 +127,94 @@ func (bitStream *opaeBitStream) validate() error {
 	return nil
 }
 
+type pciDevice struct {
+	Bus      uint64
+	Device   uint64
+	Function uint64
+}
+
+func parsePCIAddress(addr string) (*pciDevice, error) {
+	var ret = &pciDevice{}
+	pciRE := regexp.MustCompile(`^[[:xdigit:]]{4}:([[:xdigit:]]{2}):([[:xdigit:]]{2})\.([[:xdigit:]])$`)
```

Comment:
`pciRE` should be calculated once upon module initialization in the var section.